File: static/js/Violation.js

function reset_Violation(){
    $('#form_action').val('add');
    $('#Violation_id').val('');

    $('#Violation_s_date').val('');
    $('#Violation_m_date').val('');
    $('#Violation_fileNo').val('');
    $('#Violation_kootaj').val('');
    $('#Violation_cert_no').val('');
    $('#Violation_s_date_1').val('');
    $('#Violation_s_date_2').val('');
    $('#Violation_policy_1').val('');
    $('#Violation_policy_2').val('');
    $('#Violation_policy_3').val('');
    $('#Violation_policy_4').val('');
    $('#Violation_policy_5').val('');
    $('#Violation_policy_6').val('');
    $('#Violation_policy_7').val('');
    $('#Violation_policy_8').val('');
    $('#Violation_ezharname').val('');
    $('#Violation_person_type').val('');
    $('#Violation_full_name').val('');
    $('#Violation_card_no').val('');
    $('#Violation_code_no').val('');
    $('#Violation_company').val('');
    $('#Violation_ez_full_name').val('');
    $('#Violation_ez_owner').val('');
    $('#Violation_ez_agent').val('');
    $('#Violation_ez_card_no').val('');
    $('#Violation_ez_code_no').val('');
    $('#Violation_tr_full_name').val('');
    $('#Violation_tr_code_no').val('');
    $('#Violation_tr_reg_no').val('');
    $('#Violation_tr_nationality').val('');
    $('#Violation_tr_manager').val('');
    $('#Violation_tr_type').val('');
    $('#Violation_item_name').val('');
    $('#Violation_tariff_no').val('');
    $('#Violation_paid').val('');
    $('#Violation_different').val('');
    $('#Violation_fine').val('');
    $('#Violation_diff_paid').val('');
    $('#Violation_vi_type_1').val('');
    $('#Violation_vi_type_2').val('');
    $('#Violation_vi_type_3').val('');
    $('#Violation_vi_type_4').val('');
    $('#Violation_vi_type_5').val('');
    $('#Violation_vi_type_6').val('');
    $('#Violation_vi_type_7').val('');
    $('#Violation_vi_type_other').val('');
    $('#Violation_stated_tariff').val('');
    $('#Violation_stated_value').val('');
    $('#Violation_stated_weight').val('');
    $('#Violation_deducted_tariff').val('');
    $('#Violation_deducted_value').val('');
    $('#Violation_deducted_weight').val('');
    $('#Violation_law').val('');
    $('#Violation_detector').val('');
    $('#Violation_detector_other').val('');
    $('#Violation_results').val('');
    $('#Violation_commitment').val('');
    $('#Violation_receip_no').val('');
    $('#Violation_details').val('');
    $('#Violation_m_rec_date').val('');
    $('#Violation_s_rec_date').val('');
    $('#Violation_locked').val('');

}

function edit_Violation(id){
    $('#form_action').val('info');
    $('#Violation_id').val(id);
    submit_Violation();
}

function remove_Violation(id){
    $('#form_action').val('delete');
    $('#Violation_id').val(id);

    if (confirm('are you sure')){
        submit_Violation();
    }
}

function submit_Violation(){
    var f_action = $('#form_action').val()
    if (f_action != 'delete' && f_action != 'info'){
        // validation
    }
    var form = $('#form_Violation');
    $.ajax({
        url: '',
        type: 'POST',
        data: form.serialize(),
        cache: false
    }).done(function(responce){
        var d = JSON.parse(responce);
        if (d.ret == 'error' || d.action == 'error'){
            alert('Can not complete action. ' + d.ret);
            return;
        }
        switch (d.action){
            case 'add':
                var row = parseInt($('#table_Violation tr:last td:first').html());
                row = (!row)? 1 : row + 1;
                var td ='<tr id="Violation_' + d.ret + '">' +
                        '<td>' + row + '</td>' +
                        '<td id="Violation_s_date_' + d.ret + '">' + d.full_data.s_date + '</td>' +
                        '<td id="Violation_m_date_' + d.ret + '">' + d.full_data.m_date + '</td>' +
                        '<td id="Violation_fileNo_' + d.ret + '">' + d.full_data.fileNo + '</td>' +
                        '<td id="Violation_kootaj_' + d.ret + '">' + d.full_data.kootaj + '</td>' +
                        '<td id="Violation_cert_no_' + d.ret + '">' + d.full_data.cert_no + '</td>' +
                        '<td id="Violation_s_date_1_' + d.ret + '">' + d.full_data.s_date_1 + '</td>' +
                        '<td id="Violation_s_date_2_' + d.ret + '">' + d.full_data.s_date_2 + '</td>' +
                        '<td id="Violation_policy_1_' + d.ret + '">' + d.full_data.policy_1 + '</td>' +
                        '<td id="Violation_policy_2_' + d.ret + '">' + d.full_data.policy_2 + '</td>' +
                        '<td id="Violation_policy_3_' + d.ret + '">' + d.full_data.policy_3 + '</td>' +
                        '<td id="Violation_policy_4_' + d.ret + '">' + d.full_data.policy_4 + '</td>' +
                        '<td id="Violation_policy_5_' + d.ret + '">' + d.full_data.policy_5 + '</td>' +
                        '<td id="Violation_policy_6_' + d.ret + '">' + d.full_data.policy_6 + '</td>' +
                        '<td id="Violation_policy_7_' + d.ret + '">' + d.full_data.policy_7 + '</td>' +
                        '<td id="Violation_policy_8_' + d.ret + '">' + d.full_data.policy_8 + '</td>' +
                        '<td id="Violation_ezharname_' + d.ret + '">' + d.full_data.ezharname + '</td>' +
                        '<td id="Violation_person_type_' + d.ret + '">' + d.full_data.person_type + '</td>' +
                        '<td id="Violation_full_name_' + d.ret + '">' + d.full_data.full_name + '</td>' +
                        '<td id="Violation_card_no_' + d.ret + '">' + d.full_data.card_no + '</td>' +
                        '<td id="Violation_code_no_' + d.ret + '">' + d.full_data.code_no + '</td>' +
                        '<td id="Violation_company_' + d.ret + '">' + d.full_data.company + '</td>' +
                        '<td id="Violation_ez_full_name_' + d.ret + '">' + d.full_data.ez_full_name + '</td>' +
                        '<td id="Violation_ez_owner_' + d.ret + '">' + d.full_data.ez_owner + '</td>' +
                        '<td id="Violation_ez_agent_' + d.ret + '">' + d.full_data.ez_agent + '</td>' +
                        '<td id="Violation_ez_card_no_' + d.ret + '">' + d.full_data.ez_card_no + '</td>' +
                        '<td id="Violation_ez_code_no_' + d.ret + '">' + d.full_data.ez_code_no + '</td>' +
                        '<td id="Violation_tr_full_name_' + d.ret + '">' + d.full_data.tr_full_name + '</td>' +
                        '<td id="Violation_tr_code_no_' + d.ret + '">' + d.full_data.tr_code_no + '</td>' +
                        '<td id="Violation_tr_reg_no_' + d.ret + '">' + d.full_data.tr_reg_no + '</td>' +
                        '<td id="Violation_tr_nationality_' + d.ret + '">' + d.full_data.tr_nationality + '</td>' +
                        '<td id="Violation_tr_manager_' + d.ret + '">' + d.full_data.tr_manager + '</td>' +
                        '<td id="Violation_tr_type_' + d.ret + '">' + d.full_data.tr_type + '</td>' +
                        '<td id="Violation_item_name_' + d.ret + '">' + d.full_data.item_name + '</td>' +
                        '<td id="Violation_tariff_no_' + d.ret + '">' + d.full_data.tariff_no + '</td>' +
                        '<td id="Violation_paid_' + d.ret + '">' + d.full_data.paid + '</td>' +
                        '<td id="Violation_different_' + d.ret + '">' + d.full_data.different + '</td>' +
                        '<td id="Violation_fine_' + d.ret + '">' + d.full_data.fine + '</td>' +
                        '<td id="Violation_diff_paid_' + d.ret + '">' + d.full_data.diff_paid + '</td>' +
                        '<td id="Violation_vi_type_1_' + d.ret + '">' + d.full_data.vi_type_1 + '</td>' +
                        '<td id="Violation_vi_type_2_' + d.ret + '">' + d.full_data.vi_type_2 + '</td>' +
                        '<td id="Violation_vi_type_3_' + d.ret + '">' + d.full_data.vi_type_3 + '</td>' +
                        '<td id="Violation_vi_type_4_' + d.ret + '">' + d.full_data.vi_type_4 + '</td>' +
                        '<td id="Violation_vi_type_5_' + d.ret + '">' + d.full_data.vi_type_5 + '</td>' +
                        '<td id="Violation_vi_type_6_' + d.ret + '">' + d.full_data.vi_type_6 + '</td>' +
                        '<td id="Violation_vi_type_7_' + d.ret + '">' + d.full_data.vi_type_7 + '</td>' +
                        '<td id="Violation_vi_type_other_' + d.ret + '">' + d.full_data.vi_type_other + '</td>' +
                        '<td id="Violation_stated_tariff_' + d.ret + '">' + d.full_data.stated_tariff + '</td>' +
                        '<td id="Violation_stated_value_' + d.ret + '">' + d.full_data.stated_value + '</td>' +
                        '<td id="Violation_stated_weight_' + d.ret + '">' + d.full_data.stated_weight + '</td>' +
                        '<td id="Violation_deducted_tariff_' + d.ret + '">' + d.full_data.deducted_tariff + '</td>' +
                        '<td id="Violation_deducted_value_' + d.ret + '">' + d.full_data.deducted_value + '</td>' +
                        '<td id="Violation_deducted_weight_' + d.ret + '">' + d.full_data.deducted_weight + '</td>' +
                        '<td id="Violation_law_' + d.ret + '">' + d.full_data.law + '</td>' +
                        '<td id="Violation_detector_' + d.ret + '">' + d.full_data.detector + '</td>' +
                        '<td id="Violation_detector_other_' + d.ret + '">' + d.full_data.detector_other + '</td>' +
                        '<td id="Violation_results_' + d.ret + '">' + d.full_data.results + '</td>' +
                        '<td id="Violation_commitment_' + d.ret + '">' + d.full_data.commitment + '</td>' +
                        '<td id="Violation_receip_no_' + d.ret + '">' + d.full_data.receip_no + '</td>' +
                        '<td id="Violation_details_' + d.ret + '">' + d.full_data.details + '</td>' +
                        '<td id="Violation_m_rec_date_' + d.ret + '">' + d.full_data.m_rec_date + '</td>' +
                        '<td id="Violation_s_rec_date_' + d.ret + '">' + d.full_data.s_rec_date + '</td>' +
                        '<td id="Violation_locked_' + d.ret + '">' + d.full_data.locked + '</td>' +

                        '<td>' +
                        '   <span class="btn btn-warning" onclick="edit_Violation(\'' + d.ret + '\')">Edit</span>' +
                        '   <span class="btn btn-danger" onclick="remove_Violation(\'' + d.ret + '\')">Remove</span>' +
                        '</td>' +
                        '</tr>';
                $('#table_Violation tbody').append(td);
                break;
            case 'edit':
                $('#Violation_s_date_' + d.ret).html(d.full_data.s_date);
                $('#Violation_m_date_' + d.ret).html(d.full_data.m_date);
                $('#Violation_fileNo_' + d.ret).html(d.full_data.fileNo);
                $('#Violation_kootaj_' + d.ret).html(d.full_data.kootaj);
                $('#Violation_cert_no_' + d.ret).html(d.full_data.cert_no);
                $('#Violation_s_date_1_' + d.ret).html(d.full_data.s_date_1);
                $('#Violation_s_date_2_' + d.ret).html(d.full_data.s_date_2);
                $('#Violation_policy_1_' + d.ret).html(d.full_data.policy_1);
                $('#Violation_policy_2_' + d.ret).html(d.full_data.policy_2);
                $('#Violation_policy_3_' + d.ret).html(d.full_data.policy_3);
                $('#Violation_policy_4_' + d.ret).html(d.full_data.policy_4);
                $('#Violation_policy_5_' + d.ret).html(d.full_data.policy_5);
                $('#Violation_policy_6_' + d.ret).html(d.full_data.policy_6);
                $('#Violation_policy_7_' + d.ret).html(d.full_data.policy_7);
                $('#Violation_policy_8_' + d.ret).html(d.full_data.policy_8);
                $('#Violation_ezharname_' + d.ret).html(d.full_data.ezharname);
                $('#Violation_person_type_' + d.ret).html(d.full_data.person_type);
                $('#Violation_full_name_' + d.ret).html(d.full_data.full_name);
                $('#Violation_card_no_' + d.ret).html(d.full_data.card_no);
                $('#Violation_code_no_' + d.ret).html(d.full_data.code_no);
                $('#Violation_company_' + d.ret).html(d.full_data.company);
                $('#Violation_ez_full_name_' + d.ret).html(d.full_data.ez_full_name);
                $('#Violation_ez_owner_' + d.ret).html(d.full_data.ez_owner);
                $('#Violation_ez_agent_' + d.ret).html(d.full_data.ez_agent);
                $('#Violation_ez_card_no_' + d.ret).html(d.full_data.ez_card_no);
                $('#Violation_ez_code_no_' + d.ret).html(d.full_data.ez_code_no);
                $('#Violation_tr_full_name_' + d.ret).html(d.full_data.tr_full_name);
                $('#Violation_tr_code_no_' + d.ret).html(d.full_data.tr_code_no);
                $('#Violation_tr_reg_no_' + d.ret).html(d.full_data.tr_reg_no);
                $('#Violation_tr_nationality_' + d.ret).html(d.full_data.tr_nationality);
                $('#Violation_tr_manager_' + d.ret).html(d.full_data.tr_manager);
                $('#Violation_tr_type_' + d.ret).html(d.full_data.tr_type);
                $('#Violation_item_name_' + d.ret).html(d.full_data.item_name);
                $('#Violation_tariff_no_' + d.ret).html(d.full_data.tariff_no);
                $('#Violation_paid_' + d.ret).html(d.full_data.paid);
                $('#Violation_different_' + d.ret).html(d.full_data.different);
                $('#Violation_fine_' + d.ret).html(d.full_data.fine);
                $('#Violation_diff_paid_' + d.ret).html(d.full_data.diff_paid);
                $('#Violation_vi_type_1_' + d.ret).html(d.full_data.vi_type_1);
                $('#Violation_vi_type_2_' + d.ret).html(d.full_data.vi_type_2);
                $('#Violation_vi_type_3_' + d.ret).html(d.full_data.vi_type_3);
                $('#Violation_vi_type_4_' + d.ret).html(d.full_data.vi_type_4);
                $('#Violation_vi_type_5_' + d.ret).html(d.full_data.vi_type_5);
                $('#Violation_vi_type_6_' + d.ret).html(d.full_data.vi_type_6);
                $('#Violation_vi_type_7_' + d.ret).html(d.full_data.vi_type_7);
                $('#Violation_vi_type_other_' + d.ret).html(d.full_data.vi_type_other);
                $('#Violation_stated_tariff_' + d.ret).html(d.full_data.stated_tariff);
                $('#Violation_stated_value_' + d.ret).html(d.full_data.stated_value);
                $('#Violation_stated_weight_' + d.ret).html(d.full_data.stated_weight);
                $('#Violation_deducted_tariff_' + d.ret).html(d.full_data.deducted_tariff);
                $('#Violation_deducted_value_' + d.ret).html(d.full_data.deducted_value);
                $('#Violation_deducted_weight_' + d.ret).html(d.full_data.deducted_weight);
                $('#Violation_law_' + d.ret).html(d.full_data.law);
                $('#Violation_detector_' + d.ret).html(d.full_data.detector);
                $('#Violation_detector_other_' + d.ret).html(d.full_data.detector_other);
                $('#Violation_results_' + d.ret).html(d.full_data.results);
                $('#Violation_commitment_' + d.ret).html(d.full_data.commitment);
                $('#Violation_receip_no_' + d.ret).html(d.full_data.receip_no);
                $('#Violation_details_' + d.ret).html(d.full_data.details);
                $('#Violation_m_rec_date_' + d.ret).html(d.full_data.m_rec_date);
                $('#Violation_s_rec_date_' + d.ret).html(d.full_data.s_rec_date);
                $('#Violation_locked_' + d.ret).html(d.full_data.locked);

                break;
            case 'delete':
                $('tr#Violation_' + d.ret).remove();
                var i = 1;
                $('#table_Violation tbody tr').each(function(){
                    $('td:nth-child(1)',this).html(i++);
                });
                break;
			case 'info':
                $('#form_action').val('edit');
                $('#Violation_id').val(d.ret);
				
                $('#Violation_s_date').val(d.full_data.s_date);
                $('#Violation_m_date').val(d.full_data.m_date);
                $('#Violation_fileNo').val(d.full_data.fileNo);
                $('#Violation_kootaj').val(d.full_data.kootaj);
                $('#Violation_cert_no').val(d.full_data.cert_no);
                $('#Violation_s_date_1').val(d.full_data.s_date_1);
                $('#Violation_s_date_2').val(d.full_data.s_date_2);
                $('#Violation_policy_1').val(d.full_data.policy_1);
                $('#Violation_policy_2').val(d.full_data.policy_2);
                $('#Violation_policy_3').val(d.full_data.policy_3);
                $('#Violation_policy_4').val(d.full_data.policy_4);
                $('#Violation_policy_5').val(d.full_data.policy_5);
                $('#Violation_policy_6').val(d.full_data.policy_6);
                $('#Violation_policy_7').val(d.full_data.policy_7);
                $('#Violation_policy_8').val(d.full_data.policy_8);
                $('#Violation_ezharname').val(d.full_data.ezharname);
                $('#Violation_person_type').val(d.full_data.person_type);
                $('#Violation_full_name').val(d.full_data.full_name);
                $('#Violation_card_no').val(d.full_data.card_no);
                $('#Violation_code_no').val(d.full_data.code_no);
                $('#Violation_company').val(d.full_data.company);
                $('#Violation_ez_full_name').val(d.full_data.ez_full_name);
                $('#Violation_ez_owner').val(d.full_data.ez_owner);
                $('#Violation_ez_agent').val(d.full_data.ez_agent);
                $('#Violation_ez_card_no').val(d.full_data.ez_card_no);
                $('#Violation_ez_code_no').val(d.full_data.ez_code_no);
                $('#Violation_tr_full_name').val(d.full_data.tr_full_name);
                $('#Violation_tr_code_no').val(d.full_data.tr_code_no);
                $('#Violation_tr_reg_no').val(d.full_data.tr_reg_no);
                $('#Violation_tr_nationality').val(d.full_data.tr_nationality);
                $('#Violation_tr_manager').val(d.full_data.tr_manager);
                $('#Violation_tr_type').val(d.full_data.tr_type);
                $('#Violation_item_name').val(d.full_data.item_name);
                $('#Violation_tariff_no').val(d.full_data.tariff_no);
                $('#Violation_paid').val(d.full_data.paid);
                $('#Violation_different').val(d.full_data.different);
                $('#Violation_fine').val(d.full_data.fine);
                $('#Violation_diff_paid').val(d.full_data.diff_paid);
                $('#Violation_vi_type_1').val(d.full_data.vi_type_1);
                $('#Violation_vi_type_2').val(d.full_data.vi_type_2);
                $('#Violation_vi_type_3').val(d.full_data.vi_type_3);
                $('#Violation_vi_type_4').val(d.full_data.vi_type_4);
                $('#Violation_vi_type_5').val(d.full_data.vi_type_5);
                $('#Violation_vi_type_6').val(d.full_data.vi_type_6);
                $('#Violation_vi_type_7').val(d.full_data.vi_type_7);
                $('#Violation_vi_type_other').val(d.full_data.vi_type_other);
                $('#Violation_stated_tariff').val(d.full_data.stated_tariff);
                $('#Violation_stated_value').val(d.full_data.stated_value);
                $('#Violation_stated_weight').val(d.full_data.stated_weight);
                $('#Violation_deducted_tariff').val(d.full_data.deducted_tariff);
                $('#Violation_deducted_value').val(d.full_data.deducted_value);
                $('#Violation_deducted_weight').val(d.full_data.deducted_weight);
                $('#Violation_law').val(d.full_data.law);
                $('#Violation_detector').val(d.full_data.detector);
                $('#Violation_detector_other').val(d.full_data.detector_other);
                $('#Violation_results').val(d.full_data.results);
                $('#Violation_commitment').val(d.full_data.commitment);
                $('#Violation_receip_no').val(d.full_data.receip_no);
                $('#Violation_details').val(d.full_data.details);
                $('#Violation_m_rec_date').val(d.full_data.m_rec_date);
                $('#Violation_s_rec_date').val(d.full_data.s_rec_date);
                $('#Violation_locked').val(d.full_data.locked);

                break;
        }
    }).fail(function(){
        alert('Error')
    });
    reset_Violation();
}

File: views/violation.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import tornado.httpserver
import tornado.ioloop
import tornado.options
import tornado.web
from controllers.violation import Violation as Controller
from tools.redis import Redis


class Violation(tornado.web.RequestHandler):

    def get(self, *args, **kwargs):
        acc_type = Redis.get(key='acc_type')
        if acc_type == 'admin':
            permissions = Redis.get(key='permissions', type='list')
            # self.write(permissions)
            self.render('Violation.html', permissions=permissions, type='list')
        else:
            self.redirect('/login')


    def post(self, *args, **kwargs):
        id = self.get_argument('id','')
        s_date = self.get_argument('s_date', '')
        m_date = self.get_argument('m_date', '')
        file_no = self.get_argument('file_no', '')
        kootaj = self.get_argument('kootaj', '')
        cert_no = self.get_argument('cert_no', '')
        s_date_1 = self.get_argument('s_date_1', '')
        s_date_2 = self.get_argument('s_date_2', '')
        policy_1 = self.get_argument('policy_1', False)
        policy_2 = self.get_argument('policy_2', False)
        policy_3 = self.get_argument('policy_3', False)
        policy_4 = self.get_argument('policy_4', False)
        policy_5 = self.get_argument('policy_5', False)
        policy_6 = self.get_argument('policy_6', False)
        policy_7 = self.get_argument('policy_7', False)
        ezharname = self.get_argument('ezharname', 0)
        person_type = self.get_argument('person_type', False)
        full_name = self.get_argument('full_name', '')
        card_no = self.get_argument('card_no', '')
        code_no = self.get_argument('code_no', '')
        company = self.get_argument('company', '')
        ez_full_name = self.get_argument('ez_full_name', '')
        ez_owner = self.get_argument('ez_owner', '')
        ez_agent = self.get_argument('ez_agent', '')
        ez_card_no = self.get_argument('ez_card_no', '')
        ez_code_no = self.get_argument('ez_code_no', '')
        tr_full_name = self.get_argument('tr_full_name', '')
        tr_code_no = self.get_argument('tr_code_no', '')
        tr_reg_no = self.get_argument('tr_reg_no', '')
        tr_nationality = self.get_argument('tr_nationality', '')
        tr_manager = self.get_argument('tr_manager', '')
        tr_type = self.get_argument('tr_type', False)
        item_name = self.get_argument('item_name', '')
        tariff_no = self.get_argument('tariff_no', '')
        paid = self.get_argument('paid', 0)
        different = self.get_argument('different', 0)
        fine = self.get_argument('fine', 0)
        diff_paid = self.get_argument('diff_paid', 0)
        vi_type_1 = self.get_argument('vi_type_1', False)
        vi_type_2 = self.get_argument('vi_type_2', False)
        vi_type_3 = self.get_argument('vi_type_3', False)
        vi_type_4 = self.get_argument('vi_type_4', False)
        vi_type_5 = self.get_argument('vi_type_5', False)
        vi_type_6 = self.get_argument('vi_type_6', False)
        vi_type_7 = self.get_argument('vi_type_7', False)
        vi_type_other = self.get_argument('vi_type_other', '')
        stated_tariff = self.get_argument('stated_tariff', 0)
        stated_value = self.get_argument('stated_value', 0)
        stated_weight = self.get_argument('stated_weight', 0)
        deducted_tariff = self.get_argument('deducted_tariff', 0)
        deducted_value = self.get_argument('deducted_value', 0)
        deducted_weight = self.get_argument('deducted_weight', 0)
        law = self.get_argument('law', '')
        detector = self.get_argument('detector', 0)
        detector_other = self.get_argument('detector_other', '')
        results = self.get_argument('results', '')
        commitment = self.get_argument('commitment', False)
        receip_no = self.get_argument('receip_no', '')
        details = self.get_argument('details', '')
        m_rec_date = self.get_argument('m_rec_date', '')
        s_rec_date = self.get_argument('s_rec_date', '')
        locked = self.get_argument('locked', False)

        person_type = True if person_type == 1 else False
        tr_type = True if person_type == 1 else False

        Controller.add(id=id, s_date=s_date, m_date=m_date, file_no=file_no, kootaj=kootaj, cert_no=cert_no,
                       s_date_1=s_date_1, s_date_2=s_date_2, policy_1=policy_1, policy_2=policy_2, policy_3=policy_3,
                       policy_4=policy_4, policy_5=policy_5, policy_6=policy_6, policy_7=policy_7, ezharname=ezharname,
                       person_type=person_type, full_name=full_name, card_no=card_no, code_no=code_no, company=company,
                       ez_full_name=ez_full_name, ez_owner=ez_owner, ez_agent=ez_agent, ez_card_no=ez_card_no,
                       ez_code_no=ez_code_no, tr_full_name=tr_full_name, tr_code_no=tr_code_no, tr_reg_no=tr_reg_no,
                       tr_nationality=tr_nationality, tr_manager=tr_manager, tr_type=tr_type, item_name=item_name,
                       tariff_no=tariff_no, paid=paid, different=different, fine=fine, diff_paid=diff_paid,
                       vi_type_1=vi_type_1, vi_type_2=vi_type_2, vi_type_3=vi_type_3, vi_type_4=vi_type_4,
                       vi_type_5=vi_type_5, vi_type_6=vi_type_6, vi_type_7=vi_type_7, vi_type_other=vi_type_other,
                       stated_tariff=stated_tariff, stated_value=stated_value, stated_weight=stated_weight,
                       deducted_tariff=deducted_tariff, deducted_value=deducted_value, deducted_weight=deducted_weight,
                       law=law, detector=detector, detector_other=detector_other, results=results,
                       commitment=commitment, receip_no=receip_no, details=details, locked=locked)
        print('Controller.add is executed')
        # self.render('Violation.html')
        self.redirect('/violation')
        # self.write(policy_5)
        # self.write(person_type)


class ViolationData(tornado.web.RequestHandler):

    def get(self, *args, **kwargs):
        acc_type = Redis.get(key='acc_type')
        if acc_type == 'admin':
            permissions = Redis.get(key='permissions')
            self.write(permissions)
            # self.render('Violation.html', permissions=permissions, type='list')
        else:
            self.redirect('/login')



class ViolationSearch(tornado.web.RequestHandler):
    def get(self, *args, **kwargs):
        # self.write('Data')
        # self.write(str(list(Controller.find())))
        acc_type = Redis.get(key='acc_type')
        if acc_type == 'admin' or acc_type == 'user':
            permissions = Redis.get(key='permissions', type='list')
            search = ''
            fields = self.get_argument('fields', '')
            records = {}

            if fields != '':

                #To determone if it's a search or not
                search = 'search'

                fields = fields.split('|')

                perPage = self.get_argument('perPage', 10)
                page = self.get_argument('page', 1)

                fieldSearch = self.get_argument('fieldSearch', '')
                # self.write('FIELDS:')
                # self.write(fieldSearch)
                values = []
                filter = {}
                if fieldSearch != '':

                    fieldSearch = fieldSearch.split('|')

                    for item in fieldSearch:
                        value = {}
                        value['name'] = item
                        value['value'] = self.get_argument(item, '')
                        if value['value'] != '':
                            values.append(value)
                            filter[item] = self.get_argument(item, '')

                self.write('<br>VALUES:')
                self.write(str(values))
                self.write('<br>FILTER:')
                self.write(str(filter))

                records = {
                    'items': [],
                    'record_count': 30
                }

                # for x in range(0, 3):
                #     record = {}
                #     for item in fields:
                #         record[item] = 'Shit?'
                #     records['items'].append(record)

                records['items'] = Controller.find(_filter=filter)
                records['record_count'] = len(records['items'])


            FieldList = [
                {
                'name' : "file_no",
                'label' : "شماره پرونده",
                'default' : True,
                'searchable' : True,
                # 'validation' : "" - "number" - "data" - "email"
                'validation' : "number"
                },
                {
                'name' : "kootaj",
                'label' : "شماره کوتاژ",
                'default' : True,
                'searchable' : True,
                'validation' : "number"
                },
                {
                'name' : "cert_no",
                'label' : "شماره پروانه گمرکی",
                'default' : True,
                'searchable' : True,
                'validation' : "number"
                },
                {
                'name' : "company",
                'label' : "نام شرکت",
                'default' : True,
                'searchable' : True,
                'validation' : ""
                },




                {
                'name' : "card_no",
                'label' : "شماره کارت",
                'default' : True,
                'searchable' : True,
                'validation' : ""
                },
                {
                'name' : "code_no",
                'label' : "شماره کد",
                'default' : True,
                'searchable' : True,
                'validation' : ""
                },
                {
                'name' : "tariff_no",
                'label' : "شماره تعرفه",
                'default' : True,
                'searchable' : True,
                'validation' : ""
                },
                {
                'name' : "full_name",
                'label' : "نام و نام خانوادگی",
                'default' : True,
                'searchable' : True,
                'validation' : ""
                },
                ]

            user_info = {
                'user_id': '666',
                'name': 'احسان',
                'family': 'شیرزادی'
            }

            # self.write(records)
            self.render('Violation_Table.html',
                        search=search,
                        fields=fields,
                        records=records,
                        FieldList=FieldList,
                        user_info=user_info,
                        permissions=permissions
                        )
        else:
            self.redirect('/login')




    def post(self, *args, **kwargs):
        self.write('No post method allowed')
